diff --git a/init.lua b/init.lua
index 2b8af00..ac9b916 100644
--- a/init.lua
+++ b/init.lua
@@ -4,8 +4,8 @@
 local ffi = require 'ffi'
 local C, Cf
 
-local libnames = {'fftw3', 'libfftw3.so.3'}
-local flibnames = {'fftw3f', 'libfftw3f.so.3'}
+local libnames = {'libfftw3.dll'}
+local flibnames = {'libfftw3f.dll'}
 
 for i=1,#libnames do
    local ok, l = pcall(function () return ffi.load(libnames[i]) end)
