diff --git a/trepl-scm-1.rockspec b/trepl-scm-1.rockspec
index 54269b0..4a51f0b 100644
--- a/trepl-scm-1.rockspec
+++ b/trepl-scm-1.rockspec
@@ -39,11 +39,10 @@ build = {
 	     modules = {
 		    ['readline'] = {
                sources = {'readline.c'},
-               libraries = {'readline'},
+               libraries = {'libreadline'},
 			   defines = {"USE_READLINE_STATIC"},
                incdirs = {"windows"},
-               libdirs = {"windows"},
-               libraries = {'readline-win'}
+               libdirs = {"..\\..\\install"}
 			}
 		 }
 	  }
