diff --git a/optim-1.0.5-0.rockspec b/optim-1.0.5-0.rockspec
index 325e67d..7a1ce2e 100644
--- a/optim-1.0.5-0.rockspec
+++ b/optim-1.0.5-0.rockspec
@@ -21,7 +21,7 @@ dependencies = {
 build = {
    type = "command",
    build_command = [[
-cmake -E make_directory build && cd build && cmake .. -DCMAKE_BUILD_TYPE=Release -DCMAKE_PREFIX_PATH="$(LUA_BINDIR)/.." -DCMAKE_INSTALL_PREFIX="$(PREFIX)" && $(MAKE)
+cmake -E make_directory build && cd build && cmake .. -DCMAKE_BUILD_TYPE=Release -DCMAKE_PREFIX_PATH="$(LUA_BINDIR)/.." -DCMAKE_INSTALL_PREFIX="$(PREFIX)" -G "NMake Makefiles" && $(MAKE)
    ]],
    install_command = "cd build && $(MAKE) install"
 }
